components/NoteEditor/Editor/Heading.js

```diff
@@ -0,0 +1,48 @@
+import { useState, useRef, useEffect } from 'react';
+import styles from './Editor.module.scss';
+
+export default function Heading({
+  name, placeholder, value, setValue,
+}) {
+  const el = useRef();
+  const [currentHeight, setCurrentHeight] = useState();
+
+  const removeNewLine = (text) => text.replace(/\n/g, '');
+
+  const handleChange = (e) => {
+    const text = removeNewLine(e.currentTarget.value);
+
+    setCurrentHeight(0);
+    setValue(text);
+  };
+
+  useEffect(() => {
+    setCurrentHeight(el.current.scrollHeight);
+  }, []);
+
+  useEffect(() => {
+    if (currentHeight === 0) { setCurrentHeight(el.current.scrollHeight); }
+  }, [currentHeight]);
```

Comment:
el should probably also be in the dependencies, but dont quote me on that one 😄

Reply:
ESLint didn't complain, and `el` is not supposed to change

Reply:
alrighty